ksql-serde/src/main/java/io/confluent/ksql/serde/protobuf/transformer/ProtobufTransformer.java

```diff
@@ -50,26 +54,25 @@ public GenericRow convert(final MessageOrBuilder protobuf) {
     return new GenericRow(values);
   }
 
+  @SuppressWarnings("unchecked")
   private Object convertField(final Field field, final MessageOrBuilder protobuf) {
     // Get name of field
     final String name = field.name();
 
     // Find the field.
     // TODO cache this?
-    for (final Map.Entry<Descriptors.FieldDescriptor, Object> fieldEntry : protobuf.getAllFields().entrySet()) {
-      final Descriptors.FieldDescriptor fieldDescriptor = fieldEntry.getKey();
+    for (final Descriptors.FieldDescriptor fieldDescriptor : protobuf.getDescriptorForType().getFields()) {
       if (!fieldDescriptor.getName().equalsIgnoreCase(name)) {
         continue;
       }
 
       // Get the field's value.
-      final Object value = fieldEntry.getValue();
+      final Object value = protobuf.getField(fieldDescriptor);
 
       // Handle null case.
       if (value == null) {
         return null;
       }
-
       // If this field is an ENUM...
       if (fieldDescriptor.getType().name().equals("ENUM")) {
```

Comment:
I believe I need to work out moving this down into convertValue in order to support ENUM's in arrays and maps.